src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -75,42 +92,68 @@ private async Task CreateBlocksAsync()
             {
                 try
                 {
-                    // Don't mine in IBD.
-                    if (this.ibdState.IsInitialBlockDownload() && !MineDuringIBD)
+                    // Don't mine in IBD in case we are connected to any node.
+                    if (this.ibdState.IsInitialBlockDownload() && this.connectionManager.ConnectedPeers.Any())
                     {
                         int attemptDelayMs = 20_000;
                         await Task.Delay(attemptDelayMs, this.cancellation.Token).ConfigureAwait(false);
 
                         continue;
                     }
 
+                    uint timeNow = (uint) this.dateTimeProvider.GetAdjustedTimeAsUnixTimestamp();
+                    uint myTimestamp = this.slotsManager.GetMiningTimestamp(timeNow);
+
+                    if (myTimestamp < timeNow)
+                    {
+                        // Sanity check.
+                        throw new Exception("Invalid mining timestamp generated.");
+                    }
 
-                    // TODO POA check if our timestamp and if not wait for our timestamp (wait with cancellation token)
+                    uint waitingTime = myTimestamp - timeNow - 1;
```

Comment:
why -1?

Reply:
To compensate time we spend mining the block.

Reply:
How do you evaluate this to be 1?

Reply:
Just a random value. 
Better to mine slightly earlier than always be behind

Reply:
Also it compensates precision problems since unix timestamp is in seconds.

Reply:
ok fair enough.